frontend/src/hooks/useQueryString.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from 'react';
```

Comment:
not a fan of creating folder with only one file in it. Maybe you could reuse `src/templates/hooks` ?

Careful with the file name too: custom hooks should be kebab-cased